Use CUDA DSA in caffe2/operators

r-barnes opened the pull request:
Differential Revision: D42977333

malfet reviewed (approved):
LGTM from OSS PoV(caffe2 is no longer used)

r-barnes committed:
Use CUDA DSA in caffe2/operators (#95299)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/95299

Test Plan: Sandcastle

Reviewed By: ezyang, malfet

Differential Revision: D42977333

fbshipit-source-id: 24b3df0c3cca3adc6fc7fb372ed6000e2386f44e
